src/meshzoo/_helpers.py

```diff
@@ -209,3 +209,60 @@ def _compose_from_faces(corners, faces, n, edge_adjust=None, face_adjust=None):
     cells = np.concatenate(cells)
 
     return vertices, cells
+
+
+def insert_midpoints_edges(points, cells, cell_type):
+    """Collect all unique edges, calculate and return points including
+    midpoints on edges as well as the extended cells array."""
+
+    # list of supported cell types
+    supported_cell_types = ["triangle", "tetra", "quad", "hexahedron"]
+
+    if cell_type not in supported_cell_types:
+        raise TypeError("Cell type not implemented.")
+
+    number_of_points = {"triangle": 3, "tetra": 4, "quad": 4, "hexahedron": 8}
+
+    if cells.shape[1] != number_of_points[cell_type]:
+        raise ValueError("Mismatch of cell type and shape of cells array.")
+
+    number_of_edges = {"triangle": 3, "tetra": 6, "quad": 4, "hexahedron": 12}
+
+    if cell_type in ["triangle", "tetra"]:
```

Comment:
Let's separate those for every cell types. Saves us some variables and makes code more explicit.

Reply:
> This needs some tests, too. No need to get fancy, just something that shows what is happening.

added, see 10b8867

Reply:
> Let's separate those for every cell types. Saves us some variables and makes code more explicit.

done, see 72a2c39.